extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1047,10 +1047,26 @@ private static void producePersistenceUnitDescriptorFromConfig(
         descriptor.getProperties().setProperty(AvailableSettings.HBM2DDL_CHARSET_NAME,
                 persistenceUnitConfig.database.charset.name());
 
-        if (persistenceUnitConfig.database.globallyQuotedIdentifiers) {
+        // Quoting options
+        boolean globalQuoting = persistenceUnitConfig.quote.allIdentifiers
+                || persistenceUnitConfig.database.globallyQuotedIdentifiers;
+        if (globalQuoting) {
             descriptor.getProperties().setProperty(AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS, "true");
         }
 
+        if (globalQuoting && persistenceUnitConfig.quote.skipColumnDefinitions) {
+            descriptor.getProperties().setProperty(
+                    AvailableSettings.GLOBALLY_QUOTED_IDENTIFIERS_SKIP_COLUMN_DEFINITIONS, "true");
+        } else if (persistenceUnitConfig.quote.skipColumnDefinitions) {
+            LOG.warn("Enabling the quoting option 'skip-column-definitions' is useful when 'all-identifiers' is enabled too.");
+        }
+
+        if (!globalQuoting && persistenceUnitConfig.quote.autoKeywords) {
+            descriptor.getProperties().setProperty(AvailableSettings.KEYWORD_AUTO_QUOTING_ENABLED, "true");
+        } else if (persistenceUnitConfig.quote.autoKeywords) {
+            LOG.warn("Enabling the quoting option 'auto-keywords' is useful when 'all-identifiers' is disabled.");
```

Comment:
Same comment as above: it seems you should only have one property with multiple values here.

But if you feel it's necessary, I'd provide a more straightforward and with the full property keys:

```suggestion
            LOG.warnf("Enabling the quoting option '%s' is not useful because '%s' is enabled.",
                                    HibernateOrmRuntimeConfig.puPropertyKey(persistenceUnitName, "quote.auto-keywords"),
                                    HibernateOrmRuntimeConfig.puPropertyKey(persistenceUnitName, "quote.all-identifiers"));
```